monitor/scripts/ecr-check.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/sh
+
+# Dependencies: curl, jq, yq
+
+# Check the number of tags for a repository
+check_repository_tags() {
+    repository_name="${1}"
+    max_tag_count="${2}"
+    service_names="${@:3}"
+
+    # Make the API call
+    response=$(curl -s -X POST \
+                    -H "Content-Type: application/json" \
+                    -d "{\"registryAliasName\": \"bisonai\",\"repositoryName\": \"${repository_name}\"}" \
+                    https://api.us-east-1.gallery.ecr.aws/describeImageTags)
+
+    # Count the number of tags
+    available_tags=$(echo "$response" | jq '.imageTagDetails |= sort_by(.createdAt)')
+    tag_count=$(echo "$available_tags" | jq '.imageTagDetails[].imageTag' | wc -l | tr -d '[:space:]')
+
+    # Check if the tag count exceeds the maximum allowed
+    if [ "${tag_count}" -gt "${max_tag_count}" ]; then
+        echo "$repository_name: count FAIL (${tag_count} tags)"
+    else
+        echo "$repository_name: count OK"
+    fi
+
+    check_used_tags "cypress"
+    check_used_tags "baobab"
+}
+
+check_used_tags() {
+    chain="${1}"
+
+    for service_name in ${service_names}; do
+        helm_chart=$(curl -s --raw https://raw.githubusercontent.com/Bisonai/orakl-helm-charts/gcp-${chain}-prod/${service_name}/values.yaml)
+        tag=$(echo "${helm_chart}" | yq eval '.global.image.tag')
+
+        # Sometimes we split tags into listener, worker, and reporter.
+        if [ "${tag}" = "null" ]; then
+            listener_tag=$(echo "${helm_chart}" | yq eval '.global.image.listenerTag')
+            worker_tag=$(echo "${helm_chart}" | yq eval '.global.image.workerTag')
+            reporter_tag=$(echo "${helm_chart}" | yq eval '.global.image.reporterTag')
+
+            check_tag ${listener_tag}
+            check_tag ${worker_tag}
+            check_tag ${reporter_tag}
+
+            continue
+        fi
+
+        check_tag ${tag}
+    done
+}
+
+check_tag() {
+    tag="${1}"
+
```

Comment:
can we have additional parameter such as `tag_name` to print out if it is l, w, or r?

so maybe

```
check_tag() {
tag_name="$2"
....
if [ "$found_tag" = false ]; then
        echo "(${chain}) ${repository_name}: ${service_name} FAIL ($tag):($tag_name)"
    else
        echo "(${chain}) ${repository_name}: ${service_name} OK ($tag):($tag_name)"
    fi
...
}
```

Reply:
Good idea, let me add it!

Reply:
https://github.com/Bisonai/orakl/pull/1071/commits/5c685c97806eb7581cc2e6774cb3c0e28729d94f